src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,225 @@
+        //Note potential exception: Duke.Task index exceeded
+        if (!index.matches("\\d+")) {
+            throw new DukeException("☹ OOPS!!! Please specify an integer index.");
+        } else if (Integer.valueOf(index) > numTasks) {
+            throw new DukeException("☹ OOPS!!! The index specified is out of range.");
+        }
+        Task task = tasks.get(Integer.valueOf(index) - 1);
+        task.unmark();
+        return ("OK, I've marked this task as not done yet:\n  " + task);
+    }
+
+    /**
+     * Lists out the current tasks.
+     */
+    public String listTasks() {
+        String list = "Here are the tasks in your list:\n";
+        for (int i = 0; i < numTasks; i++) {
+            //Add index
+            list += (i + 1) + ".";
+            //Add task
+            list += tasks.get(i);
+            //Add nextline
+            if (i != numTasks - 1) {
+                list += "\n";
+            }
+        }
+        return list;
+    }
+
+    /**
+     * Retrieves the length of the task list.
+     *
+     * @return The length of the task list.
+     */
+    public int getLength() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Converts the TaskList to a string for writing into a file.
+     *
+     * @return The TaskList as a string.
+     */
+    public String getWrite() {
+        String s = "";
+        for (int i = 0; i < numTasks; i++) {
+            //Get task
+            Task t = tasks.get(i);
+            //Add general details
+            s += t.getType() + " - " + t.isDone() + " - " + t.getTitle();
+            //Add additional details
+            if (t instanceof Deadline) {
+                s += " - /by " + ((Deadline) t).getTime();
+            } else if (t instanceof Event) {
+                s += " - /at " + ((Event) t).getTime();
+            }
+            //Add newline
+            s += "\n";
+        }
+        return s;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof TaskList) {
+            TaskList taskList = (TaskList) o;
+            System.out.print(taskList.listTasks());
+            System.out.print(this.listTasks());
+            if (taskList.listTasks().equals(this.listTasks())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Finds tasks that match the given search input.
+     *
+     * @param input Search input.
+     * @return The list of matching tasks.
+     */
+    public String find(String input) throws DukeException {
+        String list = "Here are the matching tasks in your list:\n";
+        int found = 0;
+        for (int i = 0; i < tasks.size(); i++) {
+            if (tasks.get(i).getTitle().contains(input)) {
+                found += 1;
+                //Add index
+                list += (found) + ".";
+                //Add task
+                list += tasks.get(i);
+                //Add nextline
+                list += "\n";
```

Comment:
```suggestion
                list += String.printf("%d. %s\n", found, tasks.get(i));
```
Might be a better choice as it is more simpler.